[TimothyLawSongEn] iP

TimothyLawSongEn opened the pull request.

Commit:
Add Gradle support

TimothyLawSongEn committed:
Level-1. Added a skeletal version of Duke that starts by greeting user, echos commands, and exits on 'bye'

TimothyLawSongEn committed:
Added the ability to store tasks entered by the user and display a task list when requested.

TimothyLawSongEn committed:
Created a Task class.
Added the ability to mark tasks as done / undone.

TimothyLawSongEn committed:
Added different subclasses of Task (ToDo, Deadline, Event).

TimothyLawSongEn committed:
Added support for tracking three types of tasks.

TimothyLawSongEn committed:
Shifted files into Tasks package.

TimothyLawSongEn committed:
Edited input.txt and EXPECTED.txt for testing.

TimothyLawSongEn committed:
Revert "Edited input.txt and EXPECTED.txt for testing."

This reverts commit 5bd534c33a9f6a491e264acdf9a027e100ccc4f4.

TimothyLawSongEn committed:
Revert "Revert "Edited input.txt and EXPECTED.txt for testing.""

This reverts commit e6f1845d7da087f2278d65d8ce4bc2e4f34bfabe.

TimothyLawSongEn committed:
Added files to Tasks package.

TimothyLawSongEn committed:
Added to Exceptions package.

TimothyLawSongEn committed:
Added Exception handling in Duke to deal with errors such as incorrect inputs.

TimothyLawSongEn committed:
Update test files.

TimothyLawSongEn committed:
Add deleting of tasks from list.

TimothyLawSongEn committed:
Add a saveFormat() method that specifies format of Tasks to be saved in file.
Add isDone argument in constructor.

TimothyLawSongEn committed:
Add support for saving Tasks in file (duke.txt).

TimothyLawSongEn committed:
Update Duke's Deadline class to use dates instead of string.

TimothyLawSongEn committed:
Update test files to test for Deadline's date.

TimothyLawSongEn committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Update test files to test for Deadline's date.
  Update Duke's Deadline class to use dates instead of string.

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Tasks/Deadline.java

TimothyLawSongEn committed:
Improve OOP by abstracting out more classes. (half done)

TimothyLawSongEn committed:
Improve OOP by abstracting out more classes.

TimothyLawSongEn committed:
Change file path of Duke in test file. [In Progress]

TimothyLawSongEn committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

* commit '556af3f47a96b32898ab4cdbd65b16486a4871e8':
  Add Gradle support

TimothyLawSongEn committed:
Add Gradle support.

TimothyLawSongEn committed:
Add Test methods to Todo and Event.

ExtraShotLatte commented on src/main/java/Duke/Command/AddCommand.java:
```diff
@@ -0,0 +1,56 @@
+package Duke.Command;
+
+import Duke.Storage;
+import Duke.TaskList;
+import Duke.TaskType;
+import Duke.Tasks.Deadline;
+import Duke.Tasks.Event;
+import Duke.Tasks.ToDo;
+import Duke.Ui;
+
+import java.time.LocalDate;
```
Good use of lexical ordering for import statements

ExtraShotLatte commented on src/main/java/Duke/Command/MarkCommand.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    public void execute(TaskList l, Ui ui, Storage s) throws InvalidArgumentException {
+        if (mark == 1) {
+            l.get(index).markAsDone();
+            ui.showMark(1, l.get(index));
+        } else if (mark == 0) {
+            l.get(index).markAsDone();
+            ui.showMark(1, l.get(index));
```
Not too sure what is the use of the `int mark` here if both does the same thing

TimothyLawSongEn replied:
It was an error in my code. The second conditional statement should have been `l.get(index).markAsUndone();
            ui.showUnmark(1, l.get(index));`
Thanks for catching this error.

ExtraShotLatte commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,168 @@
+//    public static void populateArray(List<Task> l, String filePath) throws FileNotFoundException {
+//        File f = new File(filePath); // create a File for the given file path
+//        try {
+//            if(!f.exists()) {
+//                f.createNewFile();
+//            }
+//        } catch (IOException e) {
+//            // TODO Auto-generated catch block
+//            e.printStackTrace();
+//        }
+//
+//        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+//
+//        while (s.hasNext()) {
+//            String str = s.nextLine();
+//            String parts[] = str.split("~", 5);
+//
+//            switch (parts[0]) {
+//                case "T":
+//                    l.add(new ToDo(parts[2], parts[1].equals("1")));
+//                    break;
+//                case "D":
+//                    l.add(new Deadline(parts[2], parts[1].equals("1"), LocalDate.parse(parts[3])));
+//                    break;
+//                case "E":
+//                    l.add(new Event(parts[2], parts[1].equals("1"), parts[3], parts[4]));
+//                    break;
+//            }
+//        }
+//    }
+//
+//
+//    public static void save(List<Task> l, String filePath) throws IOException {
+//        FileWriter fw = new FileWriter(filePath);
+//        for (Task i : l) {
+//            fw.write(i.saveFormat() + "\n");
+//        }
+//        fw.close();
+//    }
+//}
```
May want to consider removing this chunk of comments if there is no need for it anymore.

ExtraShotLatte commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,9 @@
+package Duke;
+
+import Duke.Tasks.Task;
+
+import java.util.ArrayList;
+
+public class TaskList extends ArrayList<Task>{
+
+}
```
Perharps TaskList can be used to abstract out the insertion, deletion and manipulation of the tasks?

TimothyLawSongEn replied:
Can you elaborate? Because I feel that insertion and deletion etc can already be handled by ArrayList.

ExtraShotLatte commented on src/test/java/Duke/Tasks/EventTest.java:
```diff
@@ -0,0 +1,14 @@
+
+public class EventTest {
+    @Test
+    public void eventToString(){
+        Event e = new Event("run", true, "sun", "mon");
+        assertEquals(e.toString(), "[E][X] run (from: sun to: mon)");
+    }
```
Good use of JUnit test

ExtraShotLatte commented on tasks.txt:
```diff
@@ -0,0 +1 @@
+T~1~a
```
Not sure if adding the datafile into the repo is a good idea - ideally each local user should have their own tasklist.

TimothyLawSongEn replied:
Noted. Will delete the tasks.txt file from the repo.

ExtraShotLatte commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,58 @@
+                case "todo":
+                    return new AddCommand(TaskType.ToDo, s[1], false);
+                case "deadline":
+                    String st[] = s[1].split(" /by ", 2);
+                    return new AddCommand(TaskType.Deadline, st[0], false, LocalDate.parse(st[1]));
+                case "event":
+                    String stt[] = s[1].split(" /from ", 2);
+                    String sttt[] = stt[1].split(" /to ", 2);
```
Don't quite understand `sttt[]` perharps would be better to use names such as `stringToken1`, `stringToken2`?

TimothyLawSongEn replied:
Noted. It was bad naming on my part.

ExtraShotLatte reviewed (commented):
LGTM! Some minor confusions on the naming of the variables but understandable nonetheless.

👍 👍 👍

TimothyLawSongEn committed:
Correct Coding Standards.

TimothyLawSongEn committed:
Add JavaDoc to public methods.

TimothyLawSongEn committed:
Give users a way to find a task by searching for a keyword.

TimothyLawSongEn committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Give users a way to find a task by searching for a keyword.

jmestxr commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,66 @@
+public class Storage {
+
+    private String filePath;
+    private File f;
```
I think the name f could be renamed to something more meaningful, e.g. file, although it's quite obvious here.
Usually use scratch variables for cases like temporary storage or loop indices.

jmestxr commented on src/main/java/Duke/Tasks/DeleteCommand.java:
```diff
@@ -0,0 +1,21 @@
+
+public class DeleteCommand extends Command {
+
+    int index;
```
I think the name index could be changed to something more obvious, like taskIndex to avoid confusion.

jmestxr commented on src/main/java/Duke/Command/MarkCommand.java:
```diff
@@ -0,0 +1,43 @@
+
+public class MarkCommand extends Command {
+
+    private int mark;
```
I believe int mark here refers to whether the command is marking a task as done (`mark == 1`) or unmarking/marking as not done (`mark == 0`). In this case, since `mark` can only take two possible values, I think it is better if `mark` is a boolean variable. You could also add javadocs here explaining what `mark` means to avoid confusion.

jmestxr reviewed (commented):
Overall, I like how you followed the CS2103 Java code conventions most of the time. You could also start writing the javadocs for some classes/methods in the meantime for code that you think is less likely to be changed (e.g. for classes like Command).

TimothyLawSongEn committed:
Add GUI into Duke.

TimothyLawSongEn committed:
Improve Code Quality

TimothyLawSongEn committed:
Add asserts into Parser class

Parser code runs on certain assumptions which includes the fullCommand splitting into at most 2 sections, and switch-case will never enter default block.

Assuming the above is dangerous and might cause the program to have unexpected behaviours.

As such, asserts for the above assumptions have been added so that programmers will be instantly notified once these assumptions break.

TimothyLawSongEn committed:
Merge pull request #2 from TimothyLawSongEn/branch-A-CodeQuality

Branch a code quality

TimothyLawSongEn committed:
Merge branch 'master' of https://github.com/TimothyLawSongEn/ip

* 'master' of https://github.com/TimothyLawSongEn/ip:
  Improve Code Quality

TimothyLawSongEn committed:
Merge branch 'branch-A-Assertions' into master, resolved merge conflicts

* branch-A-Assertions:
  Add asserts into Parser class

# Conflicts:
#	src/main/java/duke/Parser.java

TimothyLawSongEn committed:
Add JavaDocs to more methods

TimothyLawSongEn committed:
Delete unused import statements

TimothyLawSongEn committed:
Add Undo function to Duke

TimothyLawSongEn committed:
Improve Code Quality after adding Undo function

TimothyLawSongEn committed:
Improve GUI by adding colours to window and circular-cropped images

TimothyLawSongEn committed:
Tweak colours and widths to improve GUI

TimothyLawSongEn committed:
Improve Code Quality by adding fullReplyMsg() to print both errorMsg and replyMsg

TimothyLawSongEn committed:
Shift fullReplyMsg() to Ui class

TimothyLawSongEn committed:
Add Ui.png to docs folder

TimothyLawSongEn committed:
Update with User Guide in GFMD

TimothyLawSongEn committed:
Merge branch 'master' of https://github.com/TimothyLawSongEn/ip

* 'master' of https://github.com/TimothyLawSongEn/ip:
  Update with User Guide in GFMD

TimothyLawSongEn committed:
Change main class to Launcher

TimothyLawSongEn committed:
Edit User Docs

TimothyLawSongEn committed:
Improve Code Quality and Add documentation

TimothyLawSongEn committed:
Change mainClassName in build.gradle
